internal/persistence/ent/client/sip.go

```diff
@@ -113,15 +139,30 @@ func (c *client) UpdateSIP(
 	}
 
 	// Save changes.
-	s, err = q.Save(ctx)
+	dbs, err = q.Save(ctx)
 	if err != nil {
 		return nil, rollback(tx, newDBError(err))
 	}
 	if err = tx.Commit(); err != nil {
 		return nil, rollback(tx, newDBError(err))
 	}
 
-	return convertSIP(s), nil
+	s := convertSIP(dbs)
+
+	// Manually get the UploaderID because the dbs result doesn't include the
+	// user edge.
+	if dbs.UploaderID != 0 {
+		u, err := c.ent.User.Query().
+			Where(user.ID(dbs.UploaderID)).
+			Select(user.FieldUUID).
+			Only(ctx)
+		if err != nil {
+			return nil, newDBErrorWithDetails(err, "update SIP: uploader not found")
+		}
+		s.UploaderID = &u.UUID
+	}
```

Comment:
Can't we get it in the initial query (L102) and save it as the `dbID` to use it here later?